drivers/platform/linux/linux_uart.c

```diff
@@ -0,0 +1,286 @@
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "error.h"
+#include "uart.h"
+#include "linux_uart.h"
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <termios.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/**
+ * @struct linux_uart_desc
+ * @brief Linux platform specific UART descriptor
+ */
+struct linux_uart_desc {
+	/** /dev/"device_id" file descriptor */
+	int fd;
+	struct termios *terminal;
+};
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Initialize the UART communication peripheral.
+ * @param desc - The UART descriptor.
+ * @param param - The structure that contains the UART parameters.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t uart_init(struct uart_desc **desc, struct uart_init_param *param)
+{
+	struct linux_uart_init_param *linux_init;
+	struct linux_uart_desc *linux_desc;
+	struct uart_desc *descriptor;
+	speed_t speed;
+	char path[64];
+
+	descriptor = malloc(sizeof(*descriptor));
+	if (!descriptor)
+		return FAILURE;
+
+	linux_desc = (struct linux_uart_desc*) malloc(sizeof(struct linux_uart_desc));
+	if (!linux_desc)
+		goto free_desc;
+
+	linux_desc->terminal = (struct termios*) malloc(sizeof(struct termios));
+	if (!linux_desc->terminal)
+		goto free_linux_desc;
+
+	descriptor->extra = linux_desc;
+	linux_init = param->extra;
+
+	snprintf(path, sizeof(path), "/dev/%s", linux_init->device_id);
```

Comment:
You should check if the string fitted in path

Reply:
Done.